### DISCUSSION TITLE
Is it possible via API to copy some of the XMP tags to EXIF of jpeg file

### DISCUSSION THREAD

**Discussion by Polpix** [immich-app/immich#14429] [Q&A] [answered]
Hello, when I set a geolocation through the web interface with the map, the position is not written in the exif of the jpeg (probably in the xmp file). When I download it, it has no geolocation, as it is not stored in the exif.
Is there a way, through the API, to copy some tags (like GPSlocation) from the xmp to the exif ?

**Comment by bo0tzz** [comment]
No, Immich never changes the original file.

**Reply by Polpix** [reply]
Thank you for your quick and clear answer.
Then, on the server itself, is it possible to modify the jpeg exif geolocation tags (with exiftool) without leading to inconsistencies in Immich, and malfunctions?

**Reply by bo0tzz** [reply, marked as answer]
Also no, Immich keeps a hash of the file in the database and if the actual file content is changed on disk, things can get weird.

**Reply by davidpcrd** [reply]
Is there something planned for this ?

**Reply by bo0tzz** [reply]
#14169

**Reply by dclobato** [reply]
> Also no, Immich keeps a hash of the file in the database and if the actual file content is changed on disk, things can get weird.

What do you mean by "get weird"?

**Reply by bo0tzz** [reply]
Any (current or future) integrity checks would be defeated, deduplication would stop working right, etc.

**Reply by dclobato** [reply]
There is no feature (current or future)  to repair the integrity of the database? Some job like "rebuild database", that will recalculate hash, verify N-M relationships (tags, e.g.)...
